internal/provider/vsphere/environ.go

```diff
@@ -118,15 +118,24 @@ func (env *environ) PrepareForBootstrap(ctx environs.BootstrapContext, controlle
 	return nil
 }
 
-// Create implements environs.Environ.
-func (env *environ) Create(ctx callcontext.ProviderCallContext, args environs.CreateParams) error {
+// ValidateModelCreation is part of the [environs.ModelResources] interface.
+func (env *environ) ValidateModelCreation(ctx context.Context) error {
+	return nil
+}
+
+// CreateModelResources is part of the [environs.ModelResources] interface.
+func (env *environ) CreateModelResources(ctx context.Context, args environs.CreateParams) error {
 	return env.withSession(ctx, func(env *sessionEnviron) error {
-		return env.Create(ctx, args)
+		return env.CreateModelResources(ctx, args)
```

Comment:
Can you change the variable name used for env? This reads like a bad recursion problem.

Reply:
k